Unable to build on Windows due to ASM_MASM check failure

**Description**

Downloaded latest release source & attempted to build on windows with crashbad backend (which appears to be default as well?) Fails with an error in crashpad's cmake config between MSVC runtime & ASM_MASM capabilities.

Building crashpad directly seems OK so it's some interaction between sentry's cmake config invoking crashpad's

**When does the problem happen**

<!-- Put an `x` where relevant. -->

- [x] During build
- [ ] During run-time
- [ ] When capturing a hard crash

**Environment**

<!-- Some issues are very OS and compiler-dependent. -->

- OS: Windows 10 Pro 64-bit version 2004
- Compiler: VS2019, MSVC 14.26.28801
- CMake config: SENTRY_BACKEND=crashpad

**Steps To Reproduce**

1. Download latest Sentry SDK source
2. Install cmake if you don't have it
3. Try `cmake -B build -D SENTRY_BACKEND=crashbad`

**Log output**
```
-- Detecting CXX compile features - done
-- SENTRY_TRANSPORT=winhttp
-- SENTRY_BACKEND=crashpad
-- The ASM_MASM compiler identification is MSVC
-- Found assembler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.26.28801/bin/Hostx64/x64/ml64.exe
-- Configuring done
CMake Error in external/crashpad/compat/CMakeLists.txt:
  MSVC_RUNTIME_LIBRARY value 'MultiThreadedDebugDLL' not known for this
  ASM_MASM compiler.


-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.
```

--- a/.github/ISSUE_TEMPLATE/bug_report.md
+++ b/.github/ISSUE_TEMPLATE/bug_report.md
@@ -24,7 +24,7 @@ assignees: ""
 
 - OS: [e.g. Windows 10, 64-bit]
 - Compiler: [e.g. MSVC 19]
-- CMake config: [e.g. SENTRY_BACKEND=inproc]
+- CMake version and config: [e.g. 3.17.2, SENTRY_BACKEND=inproc]
 
 **Steps To Reproduce**
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,7 @@
 
 **Breaking Changes**:
 
+- The minimum CMake version required to build on windows was raised to `3.16.4` to avoid potential build failures on older versions.
 - The `sentry_options_set_logger` function now accepts a `userdata` parameter.
 - The `name` parameter of `sentry_options_add_attachment(w)` was removed, it will now be inferred from the filename of `path`.
 

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,5 @@
 if(WIN32)
-	cmake_minimum_required (VERSION 3.15)
+	cmake_minimum_required (VERSION 3.16.4)
 
 	# enables support for CMAKE_MSVC_RUNTIME_LIBRARY
 	cmake_policy(SET CMP0091 NEW)

